trace_replay/trace_replay.h

```diff
@@ -46,6 +46,14 @@ enum TraceType : char {
   kBlockTraceDataBlock = 9,
   kBlockTraceUncompressionDictBlock = 10,
   kBlockTraceRangeDeletionBlock = 11,
+  // IO Trace related types based on options that will be added in trace file.
+  kIOGeneral = 12,
+  kIOFileName = 13,
```

Comment:
I'm thinking we can put all the queries into two categories: 1) file related (e.g., NewWritableFile, Sync, Close) and 2) data related (e.g., Read, Write, Append, Truncate). For all the queries, type, timestamp, filename are the standard contents in the record. The rest of the payload is depends on the query types. Otherwise, when we replay or analyze the trace in the further, we may not be able to use the trace records in a precise way.

Reply:
Yes Make sense. I was thinking in terms of that but then in IOtracer I need to check what options to write and what not because even same Functions (like GetFileSize) in different wrappers have different options (One passes filename and one doesn't). I will check once again if I can make it consistent and pass missing options that way it would be more scalable.

Reply:
Oh, I see. I did realize that the options make it more complex.